pseudopotential_calculator/calculations/bulk.py

```diff
@@ -155,44 +209,37 @@ def plot_energy_against_n_k_points(
     )
 
 
-def _get_cutoff_energy_from_calculator(
-    calculator: Castep,
-) -> float:
-    return cast(
-        float,
-        calculator.param.cut_off_energy.raw_value[0],  # type: ignore unknown
-    )
-
-
-def _plot_energy_against_cutoff_energy(
-    cutoff_energy: np.ndarray[Any, np.dtype[np.float64]],
-    energy: np.ndarray[Any, np.dtype[np.float64]],
+def plot_energy_against_cutoff_energy(
+    calculators: list[Castep],
     *,
     ax: Axes | None = None,
 ) -> tuple[Figure, Axes, Line2D]:
-    fig, ax = get_figure(ax)
+    cutoff_energy = list[float]()
+    energies = list[float]()
+    for calculator in calculators:
+        energies.append(cast(Atoms, calculator.atoms).get_potential_energy())  # type: ignore inkown
+        cutoff_energy.append(_get_cutoff_energy_from_calculator(calculator))
 
-    args = np.argsort(cutoff_energy)
-    (line,) = ax.plot(cutoff_energy[args], energy[args])  # type: ignore library
-    ax.set_xlabel("Cuttoff energy /eV")  # type: ignore library
-    ax.set_ylabel("Energy / eV")  # type: ignore library
-    ax.set_title("Plot of energy vs cutoff energy")  # type: ignore library
-    return fig, ax, line
+    return _plot_energy_against_cutoff_energy(
```

Comment:
It would be good if you could do the refactor you mentioned

Reply:
I tried to do it yesterday but I realised the code after words is almost the same length since the variable names are changed constantly. Maybe there is a way to do it with less codes?